src/bp/ui-studio/src/web/reducers/code-editor.ts

```diff
@@ -0,0 +1,31 @@
+import { CustomTemplate } from 'common/controls'
+import _ from 'lodash'
+import { handleActions } from 'redux-actions'
+import { loadInEditor, updateCode } from '~/actions'
+
+const defaultState = {
+  code: '',
+  codeCallback: undefined,
+  template: undefined
+}
+
+export interface CodeEditorReducer {
+  code: string
+  editorCallback: (code: string) => void
+  template?: string | CustomTemplate
+}
+
+const reducer = handleActions(
```

Comment:
why reducer code that is tightly related to code-editor module in studio ?

Reply:
There's a small part of the logic which must be in the studio to link the right side panel and the code editor.